tests/Codec/ArrayCodecTest.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace MongoDB\Tests\Codec;
+
+use MongoDB\Codec\ArrayCodec;
+use MongoDB\Codec\Codec;
+use MongoDB\Codec\CodecLibrary;
+use MongoDB\Codec\DecodeIfSupported;
+use MongoDB\Codec\EncodeIfSupported;
+use MongoDB\Exception\UnsupportedValueException;
+use MongoDB\Tests\TestCase;
+
+class ArrayCodecTest extends TestCase
+{
+    public function testDecodeList(): void
+    {
+        $value = [
+            'decoded',
+            'encoded',
+        ];
+
+        $this->assertSame(['decoded', 'decoded'], $this->getCodec()->decode($value));
+    }
+
+    public function testDecodeListWithGaps(): void
+    {
+        $value = [
+            0 => 'decoded',
+            2 => 'encoded',
+        ];
+
+        $this->assertSame([0 => 'decoded', 2 => 'decoded'], $this->getCodec()->decode($value));
```

Comment:
Correct, this codec only traverses the array to apply its library to each value, but since an array can be a list or a document, I decided to allow both use cases here as we don't need to care about keys in any case.